src/invoice_utils/web.py

```diff
@@ -83,20 +89,24 @@ def generate_invoice(request: InvoiceRequest):
     root_dir = Path(__file__).parent
     basic_rules = str(root_dir / "basic.json")
     engine = InvoicingEngine(basic_rules)
-    _send_mail(request)
-    return engine.process(
+    context = engine.process(
         int(request.header.number), request.header.timestamp, request.items
     )
+    renderer = PdfInvoiceRenderer("invoice")
+    invoice_path = root_dir / f"{request.header.timestamp:%Y%m%d}-{int(request.header.number):04}-invoice.pdf"
```

Comment:
It often happens that the web app is installed in a directory it does not have write access to. When that happens, the app usually allows configuring a directory with read/write access where it can perform file manipulations.
Introduce an environment variable and use the directory specified by the environment variable when one is specified.

Reply:
I created a new env variable for this. Does the responsibility for the directory to exist and have write access then fall to the one using the application or the server? So far I've created the env var and a default dir. I'm still using the root_dir path with the new directory, should I drop that and allow the directory to be anywhere on the server?